guide/protocols.md

```diff
@@ -78,23 +78,48 @@ processes), then perform any needed operations before falling back into
 the normal `gen_server` execution loop.
 
 ``` erlang
--module(my_protocol).
+-module(echo_protocol).
 -behaviour(gen_server).
 -behaviour(ranch_protocol).
 
 -export([start_link/4]).
--export([init/4]).
-%% Exports of other gen_server callbacks here.
+
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+     terminate/2, code_change/3]).
+
+-record(state, {ref, socket, transport, opts}).
+
+-define(SERVER, ?MODULE).
 
 start_link(Ref, Socket, Transport, Opts) ->
-    proc_lib:start_link(?MODULE, init, [Ref, Socket, Transport, Opts]).
+    gen_server:start_link(?MODULE, [Ref, Socket, Transport, Opts], []).
```

Comment:
You are right. To convert this to a gen_server properly, this and also the "ok = proc_lib:init_ack({ok, self()})" needs to be removed.